src/20240224_Multi_FundingUpdate/AaveV3Ethereum_FundingUpdate_20240224.sol

```diff
@@ -0,0 +1,329 @@
+import {AaveV2Ethereum, AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {AaveSwapper} from 'aave-helpers/swaps/AaveSwapper.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+
+/**
+ * @title Funding Update
+ * @author karpatkey_TokenLogic
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x4dd4dff7096bf7ab8c4c071975d40f4cf709c41b4b6b7c60777a6dd50d2ecd09
+ * - Discussion: https://governance.aave.com/t/arfc-funding-update/16675
+ */
+contract AaveV3Ethereum_FundingUpdate_20240224 is IProposalGenericExecutor {
+  using SafeERC20 for IERC20;
+
+  struct TokenToMigrate {
+    address underlying;
+    address aToken;
+    uint256 qty;
+  }
+
+  struct TokenToSwap {
+    address underlying;
+    address aToken;
+    address oracle;
+    uint256 slippage;
+  }
+
+  AaveSwapper public constant SWAPPER = AaveSwapper(MiscEthereum.AAVE_SWAPPER);
+  address public constant ALC_SAFE = 0x205e795336610f5131Be52F09218AF19f0f3eC60;
+
+  address public constant MILKMAN = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public constant PRICE_CHECKER = 0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c;
+
+  address public constant GHO_USD_FEED = 0x3f12643D3f6f874d39C2a4c9f2Cd6f2DbAC877FC;
+  address public constant GHO_ETH_FEED = 0xe3A830577f07b58895971322D5Cf65f08CCA6E02;
+
+  uint256 public constant USDC_V2_TO_MIGRATE = 300_000e6;
+
+  uint256 public constant USDC_V3_TO_SWAP = 1_250_000e6;
+  uint256 public constant USDT_V3_TO_SWAP = 1_500_000e6;
+  uint256 public constant USDT_V2_TO_SWAP = 200_000e6;
+
+  uint256 public constant GHO_ALLOWANCE = 2_900_000 ether;
+  uint256 public constant WETH_V3_ALLOWANCE = 600 ether;
+  address public constant ALLOWANCES_WALLET = 0xdeadD8aB03075b7FBA81864202a2f59EE25B312b;
+
+  function execute() external {
+    AaveV3Ethereum.COLLECTOR.approve(
+      AaveV3EthereumAssets.GHO_UNDERLYING,
+      ALLOWANCES_WALLET,
+      GHO_ALLOWANCE
+    );
+    AaveV3Ethereum.COLLECTOR.approve(
+      AaveV3EthereumAssets.WETH_A_TOKEN,
+      ALLOWANCES_WALLET,
+      WETH_V3_ALLOWANCE
+    );
+
+    // Deposit USDT into V3
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.USDT_UNDERLYING,
+      address(this),
+      IERC20(AaveV3EthereumAssets.USDT_UNDERLYING).balanceOf(address(AaveV3Ethereum.COLLECTOR))
+    );
+
+    IERC20(AaveV3EthereumAssets.USDT_UNDERLYING).forceApprove(
+      address(AaveV3Ethereum.POOL),
+      IERC20(AaveV3EthereumAssets.USDT_UNDERLYING).balanceOf(address(this))
+    );
+
+    AaveV3Ethereum.POOL.deposit(
+      AaveV3EthereumAssets.USDT_UNDERLYING,
+      IERC20(AaveV3EthereumAssets.USDT_UNDERLYING).balanceOf(address(this)),
+      address(AaveV3Ethereum.COLLECTOR),
+      0
+    );
+
+    _migrate();
+    _transfer();
+    _swap();
+  }
+
+  function _swap() internal {
```

Comment:
imo this whole functions is a bit messy:
- transfer & withdraw v3 manually
- transfer & withdraw & swap v2/v3 mix
- swap remaining v3

I think a bit cleaner to review would be:
`_withdrawV2Tokens` -> withdraw v2 tokens to SWAPPER
`_withdrawV3Tokens` -> withdraw v3 tokens to SWAPPER
manually initiate the swaps (as there's no clean way to do `unique` over two arrays.